### ISSUE TITLE
Feature idea - Keep auto-behavior but make editable?

### ISSUE THREAD

**Issue by minism** [KyleBanks/scene-ref-attribute#6]
Thanks for your work on this, its exactly what I've tried to wrangle together on my own but does it in a much better way.

One use case that comes up for me however is the following:
- I have a component which, by default, should use a child reference for a field automatically because it covers 90% of use cases.
- However, the remaining 10% of the time i need to override that reference manually.

Basically, the feature would be something like:

    [SerializeField, Child(Flag.Editable)] ...

What do you think?  I'm happy to take a stab at implementing it.

**Comment by KyleBanks** [comment]
So in this case would it still validate that the manually assigned reference must be a child? For instance, say you have 2 children that satisfy the reference, and by default it's picking #1 but you want to re-assign to #2.

Or do you expect it to allow you to assign a reference from anywhere, not necessarily a child?

**Comment by KyleBanks** [comment]
My expectation would probably be the first one, that it still needs to be a child. I need to think about this a bit more but I do think the idea has merit... 

My initial thought is to add `Flag.Editable` and still require the reference to be located on the child/parent/self. Additionally we could add something like a set of "hint" flags to the `Anywhere` attribute so it attempts to find it somewhere if null but allows editing and assigning from anywhere. 

```
[Child(Flag.Editable)] // assignable to any child
[Anywhere(Hint.Child)] // assignable to anywhere, attempts to find a child if null
```

What do you think?

**Comment by minism** [comment]
That sounds reasonable -- I'll start with the implementation of the first feature as you described since that will be sufficient for my needs too.  Hint seems good too be I don't want to overcomplicate things if its not needed.
